pkg/network/render.go

```diff
@@ -886,6 +908,110 @@ func renderIPTablesAlerter(conf *operv1.NetworkSpec, bootstrapResult *bootstrap.
 	return manifests, nil
 }
 
+// renderNetworkingConsolePlugin renders the common objects related to the networking console plugin resources
+func renderNetworkingConsolePlugin(manifestDir string, cl cnoclient.Client) ([]*uns.Unstructured, error) {
+	if !isConsolePluginCRDExist(cl) {
+		log.Printf("consoleplugins.console.openshift.io CRD does not exist yet")
+		return nil, nil
+	}
+
+	tmpFallbackConsoleImage := "quay.io/orenc/networking-console-plugin:4.16"
+	data := render.MakeRenderData()
+	data.Data["ReleaseVersion"] = os.Getenv("RELEASE_VERSION")
+
+	val, ok := os.LookupEnv("NETWORKING_CONSOLE_PLUGIN_IMAGE")
+	if ok {
+		data.Data["NetworkingConsolePluginImage"] = val
+	} else {
+		data.Data["NetworkingConsolePluginImage"] = tmpFallbackConsoleImage
+	}
+
+	manifests, err := render.RenderDir(filepath.Join(manifestDir, "networking-console-plugin"), &data)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to render networking-console-plugin manifests")
+	}
+	return manifests, nil
+}
+
+type Patch struct {
+	Type types.PatchType
+	Data []byte
+}
+
+// registerNetworkingConsolePlugin enables console plugin for networking-console if not already enabled
+func registerNetworkingConsolePlugin(cl cnoclient.Client) error {
+	if !isConsolePluginCRDExist(cl) {
+		return nil
+	}
+	pluginName := "networking-console-plugin"
+	clusterConsole := "cluster"
+	consoleKey := client.ObjectKey{Name: "cluster"}
+	consoleObj := &operv1.Console{}
+
+	crc := cl.Default().CRClient()
+	err := crc.Get(context.TODO(), consoleKey, consoleObj)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Printf("NOTICE: consoles.operator.openshift.io %s is not available yet.", clusterConsole)
+			return nil
+		}
+		return errors.Wrapf(err, "Could not get consoles.operator.openshift.io resource")
+	}
+
+	if stringInSlice(consoleObj.Spec.Plugins, pluginName) {
+		return nil
+	}
+
+	var patches []cnoclient.JsonPatch
+
+	if consoleObj.Spec.Plugins == nil {
+		patches = []cnoclient.JsonPatch{{
+			Op:    "add",
+			Path:  "/spec/plugins",
+			Value: []string{pluginName},
+		}}
+	} else {
+		patches = []cnoclient.JsonPatch{{
+			Op:    "add",
+			Path:  "/spec/plugins/-",
+			Value: pluginName,
+		}}
+	}
+
+	patchBytes, err := json.Marshal(patches)
+	patch := crclient.RawPatch(types.JSONPatchType, patchBytes)
+
+	if err != nil {
+		panic(err)
+	}
+
+	err = crc.Patch(context.TODO(), consoleObj, patch)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("registering console-plugin %q with console %q failed", pluginName, clusterConsole))
+	}
+	return nil
+}
+
 func isSupportedDualStackPlatform(platformType configv1.PlatformType) bool {
 	return dualStackPlatforms.Has(string(platformType))
 }
+
+func isConsolePluginCRDExist(cl cnoclient.Client) bool {
```

Comment:
It is created by the console-operator, but since both are cluster-operators, we want to make sure the CRD has been deployed beforehand. At cluster creation, both console-operator and CNO are running and creating resources.

Reply:
At which operator level in CVO is the console operator deployed?
Additionally to avoid unnecessary API calls please move the call to `consolePluginCRDExists` into the bootstrap part of CNO and introduce a variable that can be read in render, example:
https://github.com/openshift/cluster-network-operator/blob/6aa2f0fbf2b06ce52db1a2a039bbdbcaf588c409/pkg/platform/platform.go#L145-L149